### ISSUE TITLE
Requesting for pretrained model

### ISSUE THREAD

**Issue by iamjunwei** [beholdergan/Beholder-GAN#4]
I'm very interesting with your research "Beholder-GAN". This research can make people more beautiful. I want to test on my own datasets, and could you please provide the pretrained model? Thanks very much.

**Comment by jili5ms** [comment]
Same here. Is it possible to have a pretrained model so that we can directly consume it without any training?

**Comment by pha-nguyen** [comment]
Please download the pretrained model from here: https://github.com/tkarras/progressive_growing_of_gans
Somehow it works with me.

**Comment by travelbureau** [comment]
@pha-nguyen this did not work for me? I got further then before but i end up with a core-dump. What steps did you take? What version of Tensorflow are you running? etc. I'm running Ubuntu 18.04, TF v1.6, CUDA 10.1, GTX 1070. Perhaps not enough GPU memory?

**Comment by pha-nguyen** [comment]
@travelbureau I think so, I ran it on Tesla V100 32GB, and it consumes memory quite much. Also, check modifications in my [fork](https://github.com/pha-nguyen/Beholder-GAN). It might work for you.

**Comment by m-k-S** [comment]
Hi @pha-nguyen, thanks for debugging this code. Is there any chance you documented the steps you had to take to get this working? Thanks!

**Comment by travelbureau** [comment]
@pha-nguyen can you tell us: 

1. Which progan network pkl are you are using specifically... the link you suggest contains 100's of pkl files. I am using https://drive.google.com/file/d/1w4lO3WZP6KORp1B4QrcGuiOl5ouJC0CO/view?usp=sharing

2. What is the format of input images for beautify_image.py  e.g. png 1024x1024?

I am able to run the script, but the results are very poor. I'm trying to track down if its due to the unusualness of my input image or other factors.

Additional info... I'm occasionally getting runtime warnings such as the following: 

```
50 0.6619446873664856
[]
/Beholder-GAN/tfutil.py:1398: RuntimeWarning: Mean of empty slice.
  print(labels.mean())
/opt/conda/envs/beholder/lib/python3.6/site-packages/numpy/core/_methods.py:161: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
nan
100 0.6098600625991821
[]
```

**Comment by travelbureau** [comment]
> Hi @pha-nguyen, thanks for debugging this code. Is there any chance you documented the steps you had to take to get this working? Thanks!

Spin up a fresh deep-learning image on Google Cloud w/ V100 GPU. Here is a Dockerfile... 

```
FROM nvidia/cuda:9.0-cudnn7-runtime-ubuntu16.04

RUN apt-get update && apt-get install -y \
	wget \
	vim \
	bzip2 \
        git \
        build-essential

#Downgrade CUDA, TF issue: https://github.com/tensorflow/tensorflow/issues/17566#issuecomment-372490062
RUN apt-get install --allow-downgrades -y libcudnn7=7.0.5.15-1+cuda9.0

#Install MINICONDA
RUN wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O Miniconda.sh && \
	/bin/bash Miniconda.sh -b -p /opt/conda && \
	rm Miniconda.sh

ENV PATH /opt/conda/bin:$PATH

RUN git clone https://github.com/pha-nguyen/Beholder-GAN.git

WORKDIR Beholder-GAN

RUN mkdir models

RUN mkdir experiment_folder

RUN mkdir image_data

# You will need to download these and update their names as appropriate. 
COPY vgg_tf.pb ./models
COPY network-yourname.pkl ./experiment_folder
# Note the pkl name HAS to start with network or it will not be found... 
# Put some images in the image_data folder... 
#WORKDIR image_data
#COPY stuff .
#WORKDIR ../

ENTRYPOINT ["/bin/bash"]
```

Then `docker build -t bgan .`

Then `docker run --runtime=nvidia -it --name=bgan_container --rm bgan`

Once inside the docker follow the normal steps... I also had to comment out line 10 in `beautify_image.py` (tensorflow_hub)...

**Comment by molo32** [comment]
Can someone share a colab of this please

**Comment by ekatiyar** [comment]
Everyone here has been super helpful in getting this to work. However, the beautify_image.py file seems to only generate the latents for the image, and doesn't actually return a beautified image itself. Do any of you know how to convert these latents into the beautified image?

**Comment by ekatiyar** [comment]
Everyone here has been super helpful in getting this to work. However, I'm having issues with the labels_size argument for beautify_image.py, as the network doesn't seem to accept any labels (it asks for a size of (?, 0)). Thus, all the outputted images look the same since the beauty number doesn't do anything. @pha-nguyen I noticed your code reduced labels_size to 0 as well, so i was wondering how you got the beautification to work?

**Comment by JoshuaKnittel** [comment]
@ekatiyar I am facing the same issue. Did you get it to work somehow in the meantime?

**Comment by nguyenviettuan96** [comment]
@pha-nguyen cái file network-final.pkl có size label là 0 nhưng của beholder là 60. Làm sao bạn có thể làm cho nó chạy được. Cảm ơn bạn

**Comment by nguyenviettuan96** [comment]
> Everyone here has been super helpful in getting this to work. However, I'm having issues with the labels_size argument for beautify_image.py, as the network doesn't seem to accept any labels (it asks for a size of (?, 0)). Thus, all the outputted images look the same since the beauty number doesn't do anything. @pha-nguyen I noticed your code reduced labels_size to 0 as well, so i was wondering how you got the beautification to work?

You must train your own model by creating tfrecord dataset SCUT-FBP5500. Here is my pretrained on this dataset with label vector size 60

**Comment by tschreiner** [comment]
Hello Tuan,
can you share the pre-trained model for SCUT-FBP5500? Here or by mail to info@teddschreiner.de
